<a href="https://colab.research.google.com/github/ShuklaRajni/Airbnb-Booking-Analysis/blob/main/Airbnb_Booking_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **PROJECT TITLE - AIRBNB BOOKING ANALYSIS**

## <b> Since 2008, guests and hosts have used Airbnb to expand on traveling possibilities and present a more unique, personalized way of experiencing the world. Today, Airbnb became one of a kind service that is used and recognized by the whole world. Data analysis on millions of listings provided through Airbnb is a crucial factor for the company. These millions of listings generate a lot of data - data that can be analyzed and used for security, business decisions,understanding of customers' and providers' (hosts) behavior and performance on the platform, guiding marketing initiatives, implementation of innovative additional services and much more.</b>

## <b>This dataset has around 49,000 observations in it with 16 columns and it is a mix between categorical and numeric values. </b>

## <b> Explore and analyze the data to discover key understandings (not limited to these) such as :

* What can we learn about different hosts and areas?
* What can we learn from predictions? (ex: locations, prices, reviews, etc)
* Which hosts are the busiest and why?
* Is there any noticeable difference of traffic among different areas and what could be the reason for it? </b>


In [1]:
#import the important library files to perform action on dataframes and for plotting graphs
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.pyplot import figure
import seaborn as sns 
%matplotlib inline

In [2]:
#mounted the google drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
#read the database file from mounted google drive into dataframe df
df = pd.read_csv('/content/drive/MyDrive/Airbnb Booking Analysis-Capstone 1/Airbnb NYC 2019.csv')

In [4]:
#Display first 5 rows of the dataframe to understand the information present in the database
df.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [5]:
df_copy = df.copy()

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

In [7]:
df.isnull().sum()

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

**From Above we come to know that 10 thousand reviews are not given or missing. Also few name and hostname are missing.**

In [8]:
df.describe()

id       host_id      latitude     longitude         price  \
count  4.889500e+04  4.889500e+04  48895.000000  48895.000000  48895.000000   
mean   1.901714e+07  6.762001e+07     40.728949    -73.952170    152.720687   
std    1.098311e+07  7.861097e+07      0.054530      0.046157    240.154170   
min    2.539000e+03  2.438000e+03     40.499790    -74.244420      0.000000   
25%    9.471945e+06  7.822033e+06     40.690100    -73.983070     69.000000   
50%    1.967728e+07  3.079382e+07     40.723070    -73.955680    106.000000   
75%    2.915218e+07  1.074344e+08     40.763115    -73.936275    175.000000   
max    3.648724e+07  2.743213e+08     40.913060    -73.712990  10000.000000   

       minimum_nights  number_of_reviews  reviews_per_month  \
count    48895.000000       48895.000000       38843.000000   
mean         7.029962          23.274466           1.373221   
std         20.510550          44.550582           1.680442   
min          1.000000           0.000000           0.010000   
25%          1.000000           1.000000           0.190000   
50%          3.000000           5.000000           0.720000   
75%          5.000000          24.000000           2.020000   
max       1250.000000         629.000000          58.500000   

       calculated_host_listings_count  availability_365  
count                    48895.000000      48895.000000  
mean                         7.143982        112.781327  
std                         32.952519        131.622289  
min                          1.000000          0.000000  
25%                          1.000000          0.000000  
50%                          1.000000         45.000000  
75%                          2.000000        227.000000  
max                        327.000000        365.000000

**From above data we can say the average price of the room is around 153 and the average minimum nights stay is 7. Host listing count on a average is 7. Availabilty_365 : 113.**

## **Checked unique and Duplicates and missing values in the datasets**

In [9]:
df.nunique()

id                                48895
name                              47905
host_id                           37457
host_name                         11452
neighbourhood_group                   5
neighbourhood                       221
latitude                          19048
longitude                         14718
room_type                             3
price                               674
minimum_nights                      109
number_of_reviews                   394
last_review                        1764
reviews_per_month                   937
calculated_host_listings_count       47
availability_365                    366
dtype: int64

**No of locations are 5 and Type of rooms are 3.**

In [10]:
df.duplicated().sum()

0

# **Missing value Management**

In [11]:
#drop unnecessary Columns
df.drop(['id','name','last_review'], axis=1, inplace=True)

In [12]:
df.head()

host_id    host_name neighbourhood_group neighbourhood  latitude  \
0     2787         John            Brooklyn    Kensington  40.64749   
1     2845     Jennifer           Manhattan       Midtown  40.75362   
2     4632    Elisabeth           Manhattan        Harlem  40.80902   
3     4869  LisaRoxanne            Brooklyn  Clinton Hill  40.68514   
4     7192        Laura           Manhattan   East Harlem  40.79851   

   longitude        room_type  price  minimum_nights  number_of_reviews  \
0  -73.97237     Private room    149               1                  9   
1  -73.98377  Entire home/apt    225               1                 45   
2  -73.94190     Private room    150               3                  0   
3  -73.95976  Entire home/apt     89               1                270   
4  -73.94399  Entire home/apt     80              10                  9   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [13]:
# replace all NaN values in reviews_per_month by 0
df.reviews_per_month.fillna(0, inplace= True)
df.head()

host_id    host_name neighbourhood_group neighbourhood  latitude  \
0     2787         John            Brooklyn    Kensington  40.64749   
1     2845     Jennifer           Manhattan       Midtown  40.75362   
2     4632    Elisabeth           Manhattan        Harlem  40.80902   
3     4869  LisaRoxanne            Brooklyn  Clinton Hill  40.68514   
4     7192        Laura           Manhattan   East Harlem  40.79851   

   longitude        room_type  price  minimum_nights  number_of_reviews  \
0  -73.97237     Private room    149               1                  9   
1  -73.98377  Entire home/apt    225               1                 45   
2  -73.94190     Private room    150               3                  0   
3  -73.95976  Entire home/apt     89               1                270   
4  -73.94399  Entire home/apt     80              10                  9   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2               0.00                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [14]:
# replace all NaN values in host_name by 'no name'
df.host_name.fillna('No Name',inplace=True)
df.head()

host_id    host_name neighbourhood_group neighbourhood  latitude  \
0     2787         John            Brooklyn    Kensington  40.64749   
1     2845     Jennifer           Manhattan       Midtown  40.75362   
2     4632    Elisabeth           Manhattan        Harlem  40.80902   
3     4869  LisaRoxanne            Brooklyn  Clinton Hill  40.68514   
4     7192        Laura           Manhattan   East Harlem  40.79851   

   longitude        room_type  price  minimum_nights  number_of_reviews  \
0  -73.97237     Private room    149               1                  9   
1  -73.98377  Entire home/apt    225               1                 45   
2  -73.94190     Private room    150               3                  0   
3  -73.95976  Entire home/apt     89               1                270   
4  -73.94399  Entire home/apt     80              10                  9   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2               0.00                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [15]:
#checking for missing values 
df.isnull().any()

host_id                           False
host_name                         False
neighbourhood_group               False
neighbourhood                     False
latitude                          False
longitude                         False
room_type                         False
price                             False
minimum_nights                    False
number_of_reviews                 False
reviews_per_month                 False
calculated_host_listings_count    False
availability_365                  False
dtype: bool

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 13 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   host_id                         48895 non-null  int64  
 1   host_name                       48895 non-null  object 
 2   neighbourhood_group             48895 non-null  object 
 3   neighbourhood                   48895 non-null  object 
 4   latitude                        48895 non-null  float64
 5   longitude                       48895 non-null  float64
 6   room_type                       48895 non-null  object 
 7   price                           48895 non-null  int64  
 8   minimum_nights                  48895 non-null  int64  
 9   number_of_reviews               48895 non-null  int64  
 10  reviews_per_month               48895 non-null  float64
 11  calculated_host_listings_count  48895 non-null  int64  
 12  availability_365                

**In normal situation price of the any property cannot be 0 therefore we will exclude the data having pricing 0".**

***Good!! now that we have dropped the null values lets take a look at some statistics about our data.***

In [17]:
df=df.loc[df['price']>0]
df.describe()

host_id      latitude     longitude         price  minimum_nights  \
count  4.888400e+04  48884.000000  48884.000000  48884.000000    48884.000000   
mean   6.762203e+07     40.728953    -73.952176    152.755053        7.029887   
std    7.861666e+07      0.054532      0.046159    240.170260       20.512224   
min    2.438000e+03     40.499790    -74.244420     10.000000        1.000000   
25%    7.817310e+06     40.690100    -73.983080     69.000000        1.000000   
50%    3.079257e+07     40.723080    -73.955685    106.000000        3.000000   
75%    1.074344e+08     40.763120    -73.936290    175.000000        5.000000   
max    2.743213e+08     40.913060    -73.712990  10000.000000     1250.000000   

       number_of_reviews  reviews_per_month  calculated_host_listings_count  \
count       48884.000000       48884.000000                    48884.000000   
mean           23.271991           1.090800                        7.144628   
std            44.551331           1.597213                       32.956185   
min             0.000000           0.000000                        1.000000   
25%             1.000000           0.040000                        1.000000   
50%             5.000000           0.370000                        1.000000   
75%            24.000000           1.580000                        2.000000   
max           629.000000          58.500000                      327.000000   

       availability_365  
count      48884.000000  
mean         112.779498  
std          131.627271  
min            0.000000  
25%            0.000000  
50%           45.000000  
75%          227.000000  
max          365.000000